verification/curator-service/api/test/clients/aws-lambda-client.test.ts

```diff
@@ -81,3 +86,33 @@ describe('removePermission', () => {
         ).rejects.toThrow(expectedError);
     });
 });
+
+describe('invokeRetrieval', () => {
+    it('invoke retrieval for the given source', async () => {
+        const payload: RetrievalPayload = {
+            key: 'some key',
+            bucket: 'some bucket',
+        };
+        invokeSpy.mockResolvedValueOnce({
+            Payload: JSON.stringify(payload),
+        });
+        const res = await client.invokeRetrieval('some-source-id');
+        expect(invokeSpy).toHaveBeenCalledTimes(1);
+        expect(res).toEqual(payload);
+    });
+    it('throws when error is returned by aws api', async () => {
+        const expectedError = new Error('AWS error');
+        invokeSpy.mockRejectedValueOnce(expectedError);
+        return expect(client.invokeRetrieval('some-source-id')).rejects.toThrow(
+            expectedError,
+        );
+    });
+    it('throws when error zhen function is not run properly', async () => {
```

Comment:
nit, sp

Reply:
that new keyboard layout is killing me haha